### PR TITLE
feat: Allow user to inject inbound transport in rest cli

### LINKED ISSUE #873: Allow user to inject inbound transport in rest cli

### PR DESCRIPTION
**Pull request by fqutishat**
closes #873

Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

File: cmd/aries-agent-rest/startcmd/start.go [hunk 1]
```diff
@@ -106,6 +106,19 @@ const (
...
 	agentOutboundTransportEnvKey = "ARIESD_OUTBOUND_TRANSPORT"
+
+	agentInboundTransportEnvKey = "ARIESD_INBOUND_TRANSPORT"
+
+	agentInboundTransportFlagName = "inbound-transport"
+
+	agentInboundTransportFlagShorthand = "b"
+
+	agentInboundTransportFlagUsage = "Inbound transport type." +
+		" possible values [http] [ws]. Defaults to http if not set." +
```
Comment (DRK3) [inline cmd/aries-agent-rest/startcmd/start.go]: "possible" should be capitalized

File: docs/rest/agent_cli.md [hunk 2]
```diff
@@ -21,6 +21,7 @@ Flags:
   -r, --http-resolver-url string       HTTP binding DID resolver method and url. Values should be in method@url format. This flag can be repeated, allowing multiple http resolvers. Defaults to peer DID resolver if not set. Alternatively, this can be set with the following environment variable (in CSV format): ARIESD_HTTP_RESOLVER
   -i, --inbound-host string            Inbound Host Name:Port. This is used internally to start the inbound server. Alternatively, this can be set with the following environment variable: ARIESD_INBOUND_HOST *
   -e, --inbound-host-external string   Inbound Host External Name:Port. This is the URL for the inbound server as seen externally. If not provided, then the internal inbound host will be used here. Alternatively, this can be set with the following environment variable: ARIESD_INBOUND_HOST_EXTERNAL
+  -b, --inbound-transport string       Inbound transport type. possible values [http] [ws]. Defaults to http if not set. Alternatively, this can be set with the following environment variable: ARIESD_INBOUND_TRANSPORT
```
Comment (DRK3) [inline docs/rest/agent_cli.md]: "possible" should be capitalized

Comment (fqutishat) [reply docs/rest/agent_cli.md]: done

File: cmd/aries-agent-rest/startcmd/start.go (hunk 1 above)
Comment (fqutishat) [reply cmd/aries-agent-rest/startcmd/start.go]: done

File: test/bdd/go.sum [hunk 3]
```diff
@@ -10,13 +10,18 @@ github.com/Microsoft/hcsshim v0.8.6 h1:ZfF0+zZeYdzMIVMZHKtDKJvLHj76XCuVae/jNkjj0
 github.com/Microsoft/hcsshim v0.8.6/go.mod h1:Op3hHsoHPAvb6lceZHDtd9OkTew38wNoXnJs8iY7rUg=
 github.com/agl/ed25519 v0.0.0-20170116200512-5312a6153412 h1:w1UutsfOrms1J05zt7ISrnJIXKzwaspym5BTKGx93EI=
 github.com/agl/ed25519 v0.0.0-20170116200512-5312a6153412/go.mod h1:WPjqKcmVOxf0XSf3YxCJs6N6AOSrOx3obionmG7T0y0=
+github.com/armon/consul-api v0.0.0-20180202201655-eb2c6b5be1b6/go.mod h1:grANhF5doyWs3UAsr3K4I6qtAmlQcZDesFNEHPZAzj8=
```
Comment (troyronda) [inline test/bdd/go.sum]: I assume all these changes should happen?

File: go.sum [hunk 4]
```diff
@@ -1,27 +1,59 @@
+cloud.google.com/go v0.26.0/go.mod h1:aQUYkXzVsufM+DwF1aE+0xfcU+56JwCaLick0ClmMTw=
```
Comment (troyronda) [inline go.sum]: And these?

Comment (fqutishat) [reply go.sum]: remove it

File: test/bdd/go.sum (hunk 3 above)
Comment (fqutishat) [reply test/bdd/go.sum]: remove it not needed

**Pull request merged**